common/util/task-util.ts

```diff
@@ -161,3 +161,6 @@ export function dateMatchRepeats(
   }
   return dateMatchRepeatPattern(date, pattern);
 }
+
+export const subTasksEqual = (firstSubTask: SubTask, secondSubTask: SubTask): boolean =>
+  JSON.stringify(firstSubTask) === JSON.stringify(secondSubTask);
```

Comment:
Maybe we can use https://github.com/cornell-dti/samwise/blob/29f3126ad192279dfc18804dfd016d5716907d04/common/util/general-util.ts#L38-L56
to compare?